meson.build

```
@@ -77,8 +67,8 @@ elif target_machine.system() != 'windows'
 endif
 
 if not cc.has_function('dlopen')
-	dl_dep = cc.find_library('dl', required: not can_allow_fallback)
-	if can_allow_fallback and not dl_dep.found()
+	dl_dep = cc.find_library('dl')
```

You need to add the `required: false, disabler: true` parameters here so if the check fails (as is the case on Windows), then we explicitly request that it returns an object which has a `.found()` method that returns `false`. Try this:

```suggestion
	dl_dep = cc.find_library('dl', required: false, disabler: true)
```

---

(FWIW, this is new behaviour after Meson 0.51, from what I can see in the documentation.)

---

You don't need the disabler kwarg to make an object that has a false .found() return value. This is already the case by default, as long as it doesn't error out first because it is `required: true`.

That is only needed if you want to create an object that contaminates every function or logic gates it is used in, deleting it from reality and turning it into another disabler in its place.

Very useful if a dependency is optional but missing it means an executable cannot be built, rather than `#ifdef` out support code and build a less.capable binary.

Much less useful if you need the dependency and use .found() to check if you should fall back to another name.

---

@eli-schwartz You're right, indeed! Turns out I my understanding of disablers was not very good. The `required: false` parameter is the only one needed in the end.

---

Then , I removed the `required: true` in the `dl`  dependency. Also updated the commit log.